lsp-haskell.el

```diff
@@ -228,6 +228,15 @@ if projectile way fails"
           (user-error "Couldn't find cabal file, using: %s" dir)
         dir))))
 
+(defun lsp-haskell-execute-code-action-add-signature ()
+  "Execute code action of add signature.
+Add the type signature that GHC infers to the function located below the point."
+  (interactive)
+  (let ((action (seq-find (lambda (e) (string-prefix-p "add signature" (gethash "title" e))) (lsp-code-actions-at-point))))
```

Comment:
you should not use gethash because when lsp-use-plists is t the data structures will be plists. You may refer to `lsp-execute-code-action-by-kind` in lsp-java.

Reply:
I meant lsp-java-execute-matching-action

Reply:
Or we want the stuff in https://github.com/emacs-lsp/lsp-mode/pull/2402

Reply:
HLS *should* set an identifiable kind for the action for this reason. If it doesn't please open a PR upstream so that it does. See also https://github.com/emacs-lsp/lsp-haskell/pull/112

Reply:
thank.
I use `lsp:code-action-title`.

Reply:
Code action titles aren't the same as kinds! Titles are user facing strings that can change at any time, kinds should be more stable.

Reply:
In fact, possibly what you want is this: https://github.com/emacs-lsp/lsp-mode/blob/master/lsp-mode.el#L5520

Compare with: https://github.com/haskell/haskell-language-server/pull/1988/files

Reply:
Yes, lsp-java is using the title because the server is refusing to fix the kinds.